### DISCUSSION TITLE
Excited about using Cytoscape.js for visualization

### DISCUSSION THREAD

**Discussion by BrainAnnex** [BrainAnnex/life123#14] [Announcements]
After investigating numerous software packages to provide **visualization for networks** of reactions ("graphs" with nodes and edges), [Cytoscape.js](https://js.cytoscape.org) seems like a winner... and I started using it.

The next code release will include a Vue component to utilize these capabilities!  As well as support for it in the _Reactions_ python library, to allow convenient use from the Jupyterlab notebooks.

![network visualization](https://user-images.githubusercontent.com/41893046/178422006-e4a901b0-2609-4c0d-b68b-b41be7056cfd.PNG)

**Comment by BrainAnnex** [comment]
I'm pleased to report that the just-released **v. 1 Beta10** version contains a "Vue component" (visualization module) to display networks of reactions.  For details, please see the expanded [Visualization page](https://life123.science/visualization)
![vue_cytoscape_1](https://user-images.githubusercontent.com/41893046/179072100-e13b5ede-a8cf-4410-8a45-f95fbd3cb8c4.PNG)

The above diagram originates from [this experiment about the dynamics of two coupled reactions](https://nbviewer.org/github/BrainAnnex/life123/blob/main/experiments/1D/reaction/reaction_8.ipynb)

**Comment by mihaitodor** [comment]
I was looking recently at a graph extension for Postgres DB called Age: https://github.com/apache/age and I saw they also offer a viewer for it https://github.com/apache/age-viewer, but I haven't tried to play with it yet. Might prove useful for large graphs.

**Comment by BrainAnnex** [comment]
More recent versions of the `PyGraphVisual` library used by Life123 for the visualization of graph (network) structures, shared with the  open-source sister project [BrainAnnex.org](https://brainannex.org), offer a more polished visual presentation.  Here's the earlier example, revisited:

![group of reactions](https://github.com/user-attachments/assets/70118dd8-1e98-418a-9647-950bb28c2173)


[LIVE version of the above diagram](https://brainannex.github.io/life123/experiments/1D/reaction/reaction_8.log.htm)

[More info on the PyGraphVisual library](https://life123.science/guide)
